src/sm/networkmanager/networkmanager.cpp

```diff
@@ -0,0 +1,747 @@
+        }
+
+        mNetworkData.Set(networkID, InstanceCache());
+    }
+
+    if (auto err = mNetworkData.At(networkID).mValue.Set(instanceID, NetworkData()); !err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
+    return ErrorEnum::eNone;
+}
+
+Error NetworkManager::RemoveInstanceFromCache(const String& instanceID, const String& networkID)
+{
+    LockGuard lock {mMutex};
+
+    LOG_DBG() << "Remove instance from cache: instanceID=" << instanceID << ", networkID=" << networkID;
+
+    auto network = mNetworkData.At(networkID);
+    if (!network.mError.IsNone()) {
+        return AOS_ERROR_WRAP(network.mError);
+    }
+
+    if (auto err = network.mValue.Remove(instanceID); !err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
+    if (network.mValue.IsEmpty()) {
+        if (auto err = ClearNetwork(networkID); !err.IsNone()) {
+            return err;
+        }
+    }
+
+    return ErrorEnum::eNone;
+}
+
+Error NetworkManager::ClearNetwork(const String& networkID)
+{
+    LOG_DBG() << "Clear network: networkID=" << networkID;
+
+    StaticString<cInterfaceLen> ifName;
+
+    if (auto err = RemoveInterface(ifName.Append(cBridgePrefix).Append(networkID)); !err.IsNone()) {
+        return err;
+    }
+
+    if (auto err = FS::RemoveAll(FS::JoinPath(mCNINetworkCacheDir, networkID)); !err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
+    return AOS_ERROR_WRAP(mNetworkData.Remove(networkID));
+}
+
+Error NetworkManager::PrepareCNIConfig(const String& instanceID, const String& networkID, const NetworkParams& network,
+    NetworkConfigList& netConfigList, RuntimeConf& rtConfig, Array<StaticString<cHostNameLen>>& hosts) const
+{
+    LOG_DBG() << "Prepare CNI config: instanceID=" << instanceID << ", networkID=" << networkID;
+
+    if (auto err = PrepareHosts(instanceID, networkID, network, hosts); !err.IsNone()) {
+        return err;
+    }
+
+    netConfigList.mName    = networkID;
+    netConfigList.mVersion = cCNIVersion;
+
+    if (auto err = PrepareNetworkConfigList(instanceID, networkID, network, netConfigList); !err.IsNone()) {
+        return err;
+    }
+
+    if (auto err = PrepareRuntimeConfig(instanceID, rtConfig, hosts); !err.IsNone()) {
+        return err;
+    }
+
+    return ErrorEnum::eNone;
+}
+
+Error NetworkManager::PrepareHosts(const String& instanceID, const String& networkID, const NetworkParams& network,
+    Array<StaticString<cHostNameLen>>& hosts) const
+{
+    LockGuard lock {mMutex};
+
+    LOG_DBG() << "Prepare hosts: networkID=" << networkID;
+
+    auto retNotworkData = mNetworkData.At(networkID);
+    if (!retNotworkData.mError.IsNone()) {
+        return AOS_ERROR_WRAP(retNotworkData.mError);
+    }
+
+    auto retInstanceData = retNotworkData.mValue.At(instanceID);
+    if (!retInstanceData.mError.IsNone()) {
+        return AOS_ERROR_WRAP(retInstanceData.mError);
+    }
+
+    auto pushHostWithDomain = [&](const String& host) -> Error {
```

Comment:
Could you make it as private function? It will make this function shorter.

Reply:
Done